src/argilla/client/feedback/metrics/agreement_metrics.py

```diff
@@ -0,0 +1,273 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""This module contains metrics to gather information related to inter-Annotator agreement. """
+
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, Any, Callable, Hashable, List, Tuple, Union
+
+from nltk.metrics.agreement import AnnotationTask as NLTKAnnotationTask
+from nltk.metrics.distance import binary_distance, edit_distance, interval_distance
+from pydantic import BaseModel
+
+from argilla.client.feedback.schemas import (
+    LabelQuestion,
+    MultiLabelQuestion,
+    RankingQuestion,
+    RatingQuestion,
+    TextQuestion,
+)
+from argilla.client.feedback.schemas.enums import ResponseStatusFilter
+
+if TYPE_CHECKING:
+    from argilla.client.feedback.dataset import FeedbackDataset
+    from argilla.client.feedback.dataset.remote.dataset import RemoteFeedbackDataset
+
+
+# Expected format for the nltk's AnnotationTask
+FormattedResponses = List[Tuple[Any, Hashable, Hashable]]
+
+
+class AgreementMetricResult(BaseModel):
+    """Container for the result of an agreement metric.
+
+    It contains two fields, `metric_name` and `result` with the value of the metric.
+    """
+
+    metric_name: str
+    result: float
+
+
+def prepare_dataset_for_annotation_task(
```

Comment:
this might also be an utils thingy?

Reply:
I would also like this to be in a utils module, but it gives us some problems with circular imports due to `QUESTION_TO_DISTANCE`, plus this is quite special to this module, I don't think we should need to use this outside of this module, its quite related to the `AnnotationTask` defined in `nltk`

Reply:
lets keep it here for now.